lib/promise.js

```diff
@@ -1,6 +1,32 @@
 'use strict';
 
-const Promise = require('bluebird').getNewLibraryCopy();
+Promise.each =  async function(arr, fn) {
+   for(const item of arr) await fn(item);
+}
+
+Promise.prototype.return = func => {
+  return Promise.resolve(func);
+}
+
+Promise.prototype.tap = function(fn) {
+    return this.then(function(value) {
+         fn(value);
+         return value;
+    });
+};
+
+Promise.try = func => {
+  return new Promise((resolve, reject) => {
+  		resolve(func());
+  	});
+  }
+
+Promise.map = (fn, ctx) => {
```

Comment:
I don't know what this is suppose to be but this isn't compatible with bluebird map at all??

Reply:
I mean the code works so idk what to say lol. Try it out first then comment.